engine/src/flutter/lib/ui/painting/display_list_deferred_image_gpu_impeller.cc

```diff
@@ -163,17 +152,15 @@ void DlDeferredImageGPUImpeller::ImageWrapper::SnapshotDisplayList(
           return;
         }
 
-        wrapper->texture_registry_ = snapshot_delegate->GetTextureRegistry();
-        wrapper->texture_registry_->RegisterContextListener(
-            reinterpret_cast<uintptr_t>(wrapper.get()), weak_this);
-
+        std::shared_ptr<TextureRegistry> texture_registry =
+            snapshot_delegate->GetTextureRegistry();
         if (layer_tree) {
-          wrapper->display_list_ = layer_tree->Flatten(
+          display_list = layer_tree->Flatten(
               DlRect::MakeWH(wrapper->size_.width, wrapper->size_.height),
-              wrapper->texture_registry_);
+              texture_registry);
```

Comment:
The `texture_registry` variable can be removed and `snapshot_delegate->GetTextureRegistry()` can be passed here